lib/repo/schema.ex

```diff
@@ -1,8 +1,83 @@
 defmodule ExAudit.Schema do
-  def insert_all(module, name, schema_or_source, entries, opts) do
-    # TODO!
-    opts = augment_opts(opts)
-    Ecto.Repo.Schema.insert_all(module, name, schema_or_source, entries, opts)
+  def insert_all(module, name, schema, entries, opts) do
+    updated_opts =
+      opts
+      |> augment_opts()
+      |> maybe_add_returning_opt(name)
+
+    {:ok, result} =
+      augment_transaction(module, fn ->
+        module
+        |> do_insert_all(name, schema, entries, updated_opts)
+        |> maybe_track_changes(module, :created, schema, updated_opts)
+      end)
+
+    opts
+    |> Keyword.get(:returning)
+    |> format_result(result)
+  end
+
+  defp maybe_add_returning_opt(opts, repo) do
+    if repo.__adapter__() == Ecto.Adapters.Postgres do
```

Comment:
I don't know if there's a way to know whether an adapter supports returning other than having a hardcoded check. IIRC, the MSSQL (TDS) Adapter also supports returning data from INSERT statements.

Reply:
There is no way to check if the current adapter support returning data at runtime, I suggest we limit `insert_all` track for Postgres

Reply:
maybe would be better to make this verification here https://github.com/ZennerIoT/ex_audit/pull/63/files#diff-7a34bca522eb42f9435c431e656c3246593dc1de01810eb55efcbaa2f4040e9eR96

Reply:
If there's no way to do it, maybe we should simply investigate which adapters support it today. I think if we only check in this list, it would be fine for now: https://github.com/elixir-ecto/ecto#usage

Reply:
will do it

Reply:
@narrowtux only MSSQL supports return opt besides Postgrex. The implementation was updated to also be used with MSSQL

Reply:
@narrowtux can you take a look at the changes I did?